src/backend/distributed/commands/alter_table.c

```diff
@@ -1116,7 +1117,12 @@ CreateTableConversion(TableConversionParameters *params)
 												relation->rd_rel->relam));
 		if (!HeapTupleIsValid(amTuple))
 		{
-			ereport(ERROR, (errmsg("cache lookup failed for access method %d",
+			bool isView = get_rel_relkind(con->relationId) == RELKIND_VIEW;
```

Comment:
I don't think we need to wait this long to do this check. We could do it immediately at the start of alter_table_set_access_method.

Reply:
Moved the block and applied all review notes on the new place